### PR TITLE
Switch percentage `font-size`s to `em` and change some font sizes for consistency

### PR DESCRIPTION
**Pull request by MartijnCuppens**
We use as well percentage as em based font sizes. I've changed them to `em` font sizes and tweaked some sizes for more consistency. I also noticed setting the font-size on the subnav docs was redundant.

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -1037,7 +1041,7 @@ $thumbnail-box-shadow:              $box-shadow-sm !default;
 
 // Figures
 
-$figure-caption-font-size:          90% !default;
+$figure-caption-font-size:          $small-font-size !default;
```
Comment (XhmikosR) [inline scss/_variables.scss]: Isn't this smaller than what it was?

Comment (MartijnCuppens) [reply scss/_variables.scss]: It is, but the change it minimal (from `14.4px` to `14px`). As I said, I tweaked some font sizes to make things more consistent.

File: site/static/docs/4.3/assets/scss/_subnav.scss [hunk 2]
```diff
@@ -7,11 +7,6 @@
     opacity: .25;
   }
 
-  .dropdown-toggle {
```
Comment (mdo) [inline site/static/docs/4.3/assets/scss/_subnav.scss]: On my iPad so I can’t confirm, but was the removal of all of this intentional? Can’t recall on the `text-decoration`.

Comment (MartijnCuppens) [reply site/static/docs/4.3/assets/scss/_subnav.scss]: Yes it is. I noticed the `font-size` was redundant and the only property left in the block was `text-decoration`, which also is redundant (text-decoration is removed by `.btn`)

**Review by mdo — APPROVED** [review]

**Pull request merged**